### PR TITLE
Removed legacy from hetero plugin

### PR DESCRIPTION
**Pull request by apankratovantonp**

### CODE REVIEW

File: inference-engine/src/hetero_plugin/CMakeLists.txt [hunk 1]
```diff
@@ -16,8 +16,8 @@ ie_faster_build(${TARGET_NAME}
     UNITY
 )
 
-target_link_libraries(${TARGET_NAME} PRIVATE ade pugixml inference_engine
```
Comment (ilyachur) [inline inference-engine/src/hetero_plugin/CMakeLists.txt:19]: Just for education why did we use ade?

Comment (ilya-lavrenov) [reply inference-engine/src/hetero_plugin/CMakeLists.txt:19]: for graph splitting. Now HETERO has its own split logic

**Comment by ilya-lavrenov** [comment]
@apankratovantonp looks like we need to update new ngraph-based tests for HETERO plugins.

**Comment by ilya-lavrenov** [comment]
@apankratovantonp please, make CI green

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Replaced by https://github.com/openvinotoolkit/openvino/pull/3398

**Pull request closed without merging**